scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamFunctions.scala

```diff
@@ -22,4 +25,40 @@ object BeamFunctions {
     def processElement(c: DoFn[A, B]#ProcessContext): Unit =
       c.output(f(c.element()))
   }
+
+  case class MapSideAggregator[K, V](size: Int, semigroup: Semigroup[V]) extends DoFn[KV[K, V], KV[K, V]] {
+    var cache: SummingCache[K, V] = _
+    @StartBundle
+    def startBundle(): Unit = {
+      cache = new SummingCache[K, V](size)(semigroup)
+    }
+    @ProcessElement
+    def processElement(c: DoFn[KV[K, V], KV[K, V]]#ProcessContext): Unit = {
+      val key = c.element().getKey
+      val value = c.element().getValue
+      val evicted = cache.put(Map((key, value)))
+      evicted match {
+        case Some(m) =>
+          val mit = m.iterator
+          while (mit.hasNext) {
+            val x = mit.next()
+            c.output(KV.of(x._1, x._2))
+          }
+        case None => ()
+      }
+    }
+    @FinishBundle
+    def finishBundle(c: DoFn[KV[K, V], KV[K, V]]#FinishBundleContext): Unit = {
+      val evicted = cache.flush
+      evicted match {
+        case Some(m) =>
+          val mit = m.iterator
```

Comment:
Do we need this iterator thing here? Why not
```
m.foreach { case (k, v) => 
  c.output(KV.of(k, v), BoundedWindow.TIMESTAMP_MIN_VALUE, GlobalWindow.INSTANCE)
}
```
?

Reply:
https://github.com/twitter/scalding/pull/1942#discussion_r698812349

Reply:
Got it